src/compiler/utilities.ts

```diff
@@ -476,6 +476,38 @@ namespace ts {
         return false;
     }
 
+    /* @internal */
+    export function isDeclarationWithTypeParameters(node: Node): node is DeclarationWithTypeParameters;
+    export function isDeclarationWithTypeParameters(node: DeclarationWithTypeParameters): node is DeclarationWithTypeParameters {

```

Comment:
Or is this a trick to get the `default: staticAssertNever` to work?

---
In reply to: [135145388](https://github.com/Microsoft/TypeScript/pull/17988#discussion_r135145388) [](ancestors = 135145388)